uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/Import/Parser/BaseClass.php

```diff
@@ -0,0 +1,323 @@
+        'type' => CRM_Utils_Type::T_DATE,
+        'headerPattern' => '/End Date/i',
+      ],
+      'total_qty' => [
+        'name' => 'total_qty',
+        'title' => ts('Total Qty'),
+        'type' => CRM_Utils_Type::T_FLOAT,
+        'headerPattern' => '/Total Qty/i',
+      ],
+      'status' => [
+        'name' => 'status',
+        'title' => ts('Status'),
+        'type' => CRM_Utils_Type::T_STRING,
+        'headerPattern' => '/Status/i',
+      ],
+      'comments' => [
+        'name' => 'comments',
+        'title' => ts('Comments'),
+        'type' => CRM_Utils_Type::T_STRING,
+        'headerPattern' => '/Comments/i',
+      ],
+    ];
+  }
+
+  /**
+   * Validate a row of values in CSV based on the field type
+   *
+   * @param array $params
+   *
+   * @return string
+   */
+  private function validateFieldTypes($params){
+    $errorMessage = NULL;
+    $fields = self::getFields();
+
+    foreach($params as $key=>$value) {
+      if (isset($fields[$key]['type']) && $fields[$key]['type'] == CRM_Utils_Type::T_DATE) {
+
+        try {
+          new DateTime($value);
+        } catch (Exception $e) {
+          CRM_Contact_Import_Parser_Contact::addToErrorMsg('Invalid Date value for '.$fields[$key]['title'], $errorMessage);
+        }
+      }
+
+      if (isset($fields[$key]['type']) && ($fields[$key]['type'] == CRM_Utils_Type::T_FLOAT || $fields[$key]['type'] == CRM_Utils_Type::T_INT)) {
+        if(!is_numeric($value)) {
+          CRM_Contact_Import_Parser_Contact::addToErrorMsg('Invalid value for '.$fields[$key]['title'], $errorMessage);
+        }
+      }
+    }
+
+    return $errorMessage;
+  }
+
+  /**
+   * Validate that the Absence Type, Absence Status and any other option values in the csv row
+   * are valid option values.
+   *
+   * @param array $params
+   *
+   * @return string
+   */
+  private function validateOptionValues($params){
+    $errorMessage = NULL;
+    $absenceType = $params['absence_type'];
+    $absenceStatus = $params['status'];
+
+    if (!isset($this->absenceStatuses[$absenceStatus])) {
+      CRM_Contact_Import_Parser_Contact::addToErrorMsg('Invalid Absence Status value', $errorMessage);
+    }
+
+    if (!isset($this->absenceTypes[$absenceType])) {
+      CRM_Contact_Import_Parser_Contact::addToErrorMsg('Invalid Absence type value', $errorMessage);
+    }
+
+    return $errorMessage;
+  }
+
+  /**
+   * Retrieve available Absence types from the Absence Type table
+   * Also for an absence type that allows accrual request, because the CSV imported does not
+   * specify the request type but rather the absence type is listed with '(Credit) in front
+   * to show that it is an accrual request. This is also simulated when creating the absence Types array to
+   * to match what is in the CSV.
+   *
+   * @return array
+   */
+  private function getAbsenceTypes() {
+    $absenceType = new AbsenceType();
+    $absenceType->is_active = 1;
+    $absenceType->find();
```

Comment:
The AbsenceType BAO has a `getEnabledAbsenceTypes` method that does this same thing